### PR TITLE
haskell package set configuration clean-up

### PR DESCRIPTION
**Pull request by mpickering**
###### Motivation for this change

I went through all the overrides in the `configuration-common.nix` and `configuration-ghc-8.2.nix` files and checked whether they were still necessary. 

I tested each attribute with `nix-shell -p "haskellPackages.<pkg>"` to check it still built correctly with `ghc822`. 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 1]
```diff
@@ -49,7 +49,10 @@ self: super: {
   Dust-crypto = dontCheck super.Dust-crypto;
   hasql-postgres = dontCheck super.hasql-postgres;
   hspec = super.hspec.override { stringbuilder = dontCheck self.stringbuilder; };
-  hspec-core = super.hspec-core.override { silently = dontCheck super.silently; temporary = dontCheck super.temporary; };
+  hspec-core = dontCheck (super.hspec-core.override { silently = dontCheck self.silently; });
```
Comment (peti) [inline pkgs/development/haskell-modules/configuration-common.nix]: This change doesn't look right. The original version did run the `hspec-core` test suite and it needed special untested variants of some dependencies to accomplish that without creating an infinite loop during evaluation. The new code does not run test suite, however, and therefore it won't need the untested version of `silently`. I would suggest to either (a) enable tests of `hspec-core` (which is what I'd prefer) or (b) to remove the override that passes a special version of `silently`.

**Review by peti — COMMENTED** [review]
Wow, very cool! Thank you very much for the effort you've put into this clean-up. 😃

I have one question about the change to `hspec-core`. Once we've figured that out; I'll merge the change to  the haskell-updates branch for testing on Hydra.

**Comment by mpickering** [comment]
I just reverted the change to `hspec-core`. The tests fail on `HEAD` but let's not worry about that for now.

**Comment by peti** [comment]
One more minor nit: can you please re-base this PR for the current `master`?

**Comment by mpickering** [comment]
Rebased now.

**Comment by peti** [comment]
Building at https://hydra.nixos.org/eval/1424581.

**Review by peti — APPROVED** [review]

**Comment by mpickering** [comment]
Looks like I didn't break anything if I understand the output correctly?

**Pull request closed without merging**